src/controllers/Controller.java

```diff
@@ -54,107 +51,102 @@ public void initialize(URL location, ResourceBundle resources) {
+
+        for (FoodItemModel foodItemModel : foodItemModelArrayList) {
+            final Button tempButton = new Button(foodItemModel.getName());
+            tempButton.setMinHeight(69);
+            tempButton.setMinWidth(159);
 
-        for (FoodItemModel foodItemModel : foodItemsList) {
+            foodItemPane.getChildren().addAll(tempButton);
+            tempButton.setOnAction(new EventHandler<ActionEvent>() {
+                @Override
+                public void handle(ActionEvent event) {
+                    String selectedItem = tempButton.getText();
 
-            observableList.add(foodItemModel.getName());
+                    addItemToReceipt(selectedItem, selectedCategory, categoryModelHashMap);
+                }
+            });
         }
-        foodItemsListView.setItems(observableList);
     }
 
-    private void initializeReceiptTable() {
-
-        foodItemsListView.setOnMouseClicked(new EventHandler<MouseEvent>() {
-            @Override
-            public void handle(MouseEvent mouseEvent) {
+    private void addItemToReceipt(String selectedItem, String categoryName, HashMap<String, CategoryModel> categoryModelHashMap) {
 
-                String selectedFoodName = foodItemsListView.getSelectionModel().getSelectedItem().toString();
-                String categoryName = categoryListView.getSelectionModel().getSelectedItem().toString();
-                CategoryModel categoryModel = categoryModelHashMap.get(categoryName);
+        FoodItemsService foodItemsService = new FoodItemsService();
+        ArrayList<FoodItemModel> foodItemModelArrayList = foodItemsService.get(categoryName);
+        CategoryModel categoryModel = categoryModelHashMap.get(categoryName);
 
-                ArrayList<FoodItemModel> foodItem = categoryModel.getFoodItemsList();
-                for (FoodItemModel foodItemModel : foodItem) {
-                    if (selectedFoodName.equals(foodItemModel.getName())) {
+        for (FoodItemModel foodItemModel : foodItemModelArrayList) {
+            if (selectedItem.equals(foodItemModel.getName())) {
 
-                        FoodItemModel foodItemsList = categoryModel.find(selectedFoodName);
-                        selectedFoodItemsToDisplay.addAll(foodItemsList);
+                FoodItemModel foodItemsList = categoryModel.find(selectedItem);
+                selectedFoodItemsToDisplay.addAll(foodItemsList);
 
-                        Double individualItemPrice = Double.valueOf(foodItemModel.getPrice());
-                        receipt.updateTotal(individualItemPrice);
-                    }
-                }
-                taxField.setText(String.valueOf(receipt.getTax()));
-                totalField.setText(String.valueOf(receipt.getGrandTotal()));
-                receiptTableView.setItems(selectedFoodItemsToDisplay);
+                Double individualItemPrice = Double.valueOf(foodItemModel.getPrice());
+                receipt.updateTotal(individualItemPrice);
```

Comment:
💡 this is kind of a carry over comment from the other PR.  I would rather see that you add the food item to the list of items purchased in the receipt object and then when you want the total you just ask the receipt object for the total while passing in nothing to the function because the receipt knows everything that you want to purchase and can calculate the total for you dynamically every time.